helix-lsp/src/lib.rs

```diff
@@ -204,6 +204,20 @@ pub mod util {
         // in reverse order.
         edits.sort_unstable_by_key(|edit| edit.range.start);
 
+        // Generate a diff if the edit is a full document replacement.
+        #[allow(clippy::collapsible_if)]
+        if edits.len() == 1 {
+            let is_document_replacement = edits.first().and_then(|edit| {
```

Comment:
As clippy is telling us here, you don't need this intermediate variable.

Reply:
The clippy warning is for turning it into one `if` statement. Clippy actually tells you to use an intermediate statement.

Reply:
Looks like two clippy rules are fighting each other, [`collapsible_if`](https://rust-lang.github.io/rust-clippy/rust-1.64.0/index.html#collapsible_if) and [`block_in_if_conditions`](https://rust-lang.github.io/rust-clippy/rust-1.64.0/index.html#blocks_in_if_conditions). I think this is the lesser of the two evils here readability-wise 👍